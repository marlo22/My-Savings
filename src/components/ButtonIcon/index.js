import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { TouchableOpacity } from 'react-native';
import { Icon } from 'native-base';

import { primaryColor } from '../../themes';

const StyledIcon = styled(Icon)`
  color: ${primaryColor};
  font-size: 22px;
  margin-horizontal: 10px;
`;

const ButtonIcon = ({ onPress, type = "AntDesign", name, buttonProps = {}, iconProps = {} }) => (
  <TouchableOpacity onPress={onPress} {...buttonProps}>
    <StyledIcon
      type={type}
      name={name}
      {...iconProps}
    />
  </TouchableOpacity>
);

ButtonIcon.propTypes = {
  onPress: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  buttonProps: PropTypes.object,
  iconProps: PropTypes.object
};

export default ButtonIcon;