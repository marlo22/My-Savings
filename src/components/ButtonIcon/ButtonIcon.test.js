import React from 'react';
import { shallow } from 'enzyme';

import ButtonIcon from './';

const onPress = jest.fn();
const type = 'AntDesign';
const name = 'user';
const buttonProps = { style: { color: 'red' } };
const iconProps = { style: { fontSize: 20 } };

const component = shallow(
  <ButtonIcon
    onPress={onPress}
    type={type}
    name={name}
    buttonProps={buttonProps}
    iconProps={iconProps}
  />
);

describe('ButtonIcon', () => {
  test('should render without errors', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('TouchableOpacity', () => {
    test('click on TouchableOpacity should call onPress callback', () => {
      const touchableOpacity = component.find('TouchableOpacity');
      touchableOpacity.simulate('press');
      expect(onPress).toHaveBeenCalled();
    });
  
    test('should set passed button props', () => {
      const touchableOpacity = component.find('TouchableOpacity');
      expect(touchableOpacity.prop('style')).toEqual(buttonProps.style);
    });
  });

  describe('Icon', () => {
    const iconDisplayName = 'Styled(Styled(Icon))';

    test('should set passed icon type', () => {
      const icon = component.find(iconDisplayName);
      expect(icon.prop('type')).toEqual(type);
    });

    test('should set passed icon name', () => {
      const icon = component.find(iconDisplayName);
      expect(icon.prop('name')).toEqual(name);
    });

    test('should set passed icon props', () => {
      const icon = component.find(iconDisplayName);
      expect(icon.prop('style')).toEqual(iconProps.style);
    });
  });
});