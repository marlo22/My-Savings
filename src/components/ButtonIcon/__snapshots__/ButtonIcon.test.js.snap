// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "color": "red",
    }
  }
>
  <Styled(Styled(Icon))
    name="user"
    style={
      Object {
        "fontSize": 20,
      }
    }
    type="AntDesign"
  />
</TouchableOpacity>
`;
